packages/docmanager/test/dialog.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { DocumentRegistry } from '@jupyterlab/docregistry';
+import { Contents, ServiceManager } from '@jupyterlab/services';
+import { Widget } from '@lumino/widgets';
+import * as Mock from '@jupyterlab/testutils/lib/mock';
+import { DocumentManager, renameFile } from '../src';
+
+function shouldOverwriteFalse(path: string): Promise<boolean> {
+  return Promise.resolve(false);
+}
```

Comment:
We are using Jest for testing. The documentation for mocking is [there](https://jestjs.io/docs/26.x/mock-functions#mocking-partials).

Something like the following should to the trick

```suggestion
import { DocumentManager, renameFile } from '../src';

jest.mock('../src', () => {
  const originalModule = jest.requireActual('../src');

  //Mock the shouldOverwrite
  return {
    __esModule: true,
    ...originalModule,
    shouldOverwrite: jest.fn().mockReturnValue(false),
  };
});
```

Reply:
Thanks, I didn't know about mocking partials, which seems to be very useful. But in this case it doesn't work, because it only mocks shouldOverwrite if you call it from the test. But it's called from within `dialog.ts` and stays unmocked there. A lot of people seem to have run [into this](https://github.com/facebook/jest/issues/936) (even found a comment from myself 2 years ago in this thread 😬).
 
 I tried [option 1](https://github.com/facebook/jest/issues/936#issuecomment-545080082) but ended up with the typescript error message `TypeError: Cannot redefine property: shouldOverwrite`.

Reply:
Thanks for trying that path @ephes 

I look a bit more at the code of `shouldOverwrite`. And as this is triggering a dialog, you could make use of the test helper `dimissDialog` (as you are emulating a rejection). See that example:

https://github.com/jupyterlab/jupyterlab/blob/43f784a35b5ec474f5a94433c5d2d4d6d712cbc8/packages/apputils/test/sessioncontext.spec.ts#L578

Something like the following code should do the trick:

```ts
        await Promise.all([dismissDialog(), renameFile(
          manager,
          'foo.ipynb',
          'bar.ipynb'
        )]).catch(error => {
          // error should be 'File not renamed'
          expect(error).not.toBe(alreadyExistsError);
        });
```

Reply:
Oh wow, this is much better.